### PR TITLE
Summer 2023

### PR DESCRIPTION
**Pull request by countvajhula**
DIA conducted in August 2023.

### CODE REVIEW

**Comment by benknoble** [comment]
Will this branch eventually get the reconciled analyses, too?

**Comment by countvajhula** [comment]
@benknoble Yes, looks like I accidentally closed this PR earlier. The plan would be to merge the reconciled analyses (#8) next week when we complete it for ideas.

File: evaluation/antecedents/antecedents.md
Comment (benknoble) [inline evaluation/antecedents/antecedents.md]: I think you'll want to change this to `#lang abe/antecedents2 …`

Comment (countvajhula) [reply evaluation/antecedents/antecedents.md]: Fixed, thanks!

**Comment by countvajhula** [comment]
Okay, I believe all the data is in and ready for tallying @benknoble 🙂

**Comment by benknoble** [comment]
@countvajhula I'm getting an error about the attributions in antecedents and ideas:
```
λ racket evaluation/antecedents/antecedents.md
validate-attributions: bad attributions: #hash((ABEFoundation . 28) (Africa Greentec . 99/10) (Airbnb . 6/7) (Amazon . 9/4) (BitBucket . 3/4) (CircleCI . 3/4) (Etsy . 9/4) (GitHub . 3/4) (GitLab . 3/4) (Grailed . 9/10) (Gusto . 18/5) (IMVU . 3/4) (MasterCard . 6/7) (NASDAQ . 9/10) (NYSE . 9/10) (OpenFare . 18/5) (Patreon . 6/7) (Paypal . 6/7) (Quickbooks . 18/5) (RoyaltyOffRevenueLicense . 33/10) (SourceCred . 3/2) (SourceHut . 3/4) (Square . 18/5) (Stripe . 6/7) (TravisCI . 3/4) (Uber . 729/140) (Vinted . 9/10) (Visa . 6/7) (eBay . 9/10) (ebay . 9/5))
  context...:
   body of "/Users/Knoble/abe/dia-old-abe/evaluation/antecedents/antecedents.md"
```

I didn't look too deeply because I don't recall the exact process here. Branch for reproduction is https://github.com/benknoble/dia-old-abe/tree/bk-tally.

What values are the attributions (`hash-values`) here suppose to represent? Percentages?

**Comment by countvajhula** [comment]
@benknoble Yeah, looks like it's percentages. The problem seems to be the missing antecedents on some of the leaf nodes. Those leaf ideas represent some percentage of the total allocation to ideas, and since they contain no antecedents to receive that allocation, it's coming up shy of 100%. Adding a dummy antecedent brings it up to 100%.

I think we had decided that antecedents should actually happen as a separate dialectical process that curates ideas and antecedents globally across all projects, and that each individual project would simply be "adding" antecedents for each idea as evident from the unique perspective of the project. For example (to jog our memory and think through the implications more), for "equity" the antecedents here include "SourceCred" and "RoyaltyOffRevenueLicense" which are adjacent to Old Abe. But a different project in investment banking (say) might see some finance think tank, or even something like Y combinator, as an antecedent. Each project could be seen as a unique perspective on a common idea and each instance of DIA adds to that collective record of ideas and antecedents.

Some unanswered questions there are about syntax and semantics. "Equity" has many distinct meanings. But ideas are not purely syntax, so we want the ideas representation to be disambiguated by meaning. Probably each idea could have a name as well as a definition / description, so that DIA for projects would be able to link to the correct ideas (and that's all they really need to do, but it would be good if they also supplied any apparent antecedents since it would be deduplicated anyway).

All that is to say, this approach would mean that it's OK for ideas not to report (even any) antecedents, as every project pays attributive revenue up to all of the collective antecedents rather than just the ones it reports. For our purposes right now, we could add a placeholder antecedent to each idea, like:

`Money [MoneyAntecedents]`

Wdyt? I've created a PR with this change.

I think eventually all antecedents would be treated using such placeholders, instead of being named individually by projects. And the "ideas and antecedents" process would receive revenues from all projects, indexed by idea, and would be tasked with paying the antecedents.

**Comment by benknoble** [comment]
It's all a bit over my head but I think see where you're going—at any rate, I understand the problem and solution. PR link?

**Comment by countvajhula** [comment]
It's against your fork: https://github.com/benknoble/dia-old-abe/pull/1

**Comment by benknoble** [comment]
Darned forks… GitHub doesn't subscribe you to forks (I was not "watching" it) by default, which is frankly a flaw; it actively discourages forking and DVCS models that Git was designed around. Net result: I didn't even know the PR had been made, and wouldn't have without your note or checking the repo carefully. Thanks for the link.

**Comment by countvajhula** [comment]
Yeah, for sure. I forked some Emacs projects once and got the former maintainer's blessing to continue as the new maintainer, but I had to jump through hoops with GitHub support so that the repos wouldn't have the second class treatment that forks have (e.g. commits to the repo weren't showing up in my timeline - which, I don't know why forks should be treated differently here...). In fact, I found the support ticket in my email and this was my note to them:

> If I may make a suggestion here -- I would say that forked repositories should not be considered second-class citizens (for instance, wrt contributions and tracking). I'd love it if this "detaching" was simply not necessary, since I'd ideally have preferred to preserve the association with the upstream repo, even though my fork is going to be the canonical one going forward. There's no shame in being a fork, but rather, there is pride at collaborating with others -- you must not discourage this by treating forks as "lesser" and by the same token encourage users to create independent derivative works without making an attempt to collaborate.

> Just my 2c, in case other users have suggested the same thing, I hope you will take it into consideration and forward it to the appropriate product folks.

(Of course, these things wouldn't happen in an ABE world where we not only have access to the source but get paid to improve the product in any way (including by creating forks of services like GitHub 🙂 ).)

**Comment by benknoble** [comment]
Wow, you went to a lot of effort ! I have left some of my forks as forks, but now I'm considering trying to detach them to get "first-classness." I think there's a method to do so in Settings. 

Anyway, tallying should be taken care of now. PR to this repo submitted.

**Comment by countvajhula** [comment]
@benknoble At least back when I did it, there was no option in settings and you had to contact Support directly. They were very responsive though and it was done in a matter of minutes as I recall (but various second-classness issues persisted for months afterward unfortunately. I think they've fixed that now).

And thanks for doing the tallying!!!

**Review by jairtrejo — APPROVED** [review]

**Review by benknoble — APPROVED** [review]

**Comment by countvajhula** [comment]
LGTM

**Comment by countvajhula** [comment]
Robin says, "just reviewed the document, and everything looks good!"

**Review by apromessi — APPROVED** [review]

**Pull request merged**
